CONTRIBUTING.rst

```diff
@@ -0,0 +1,234 @@
+=================
+How to contribute
+=================
+
+First, thanks for considering contributing to Astrality, that means a lot!
+Here we describe how you can help out, either by improving the documentation, submitting issues, or commit pull requests.
+
+If you end up contributing, please consider adding yourself to the file ``CONTRIBUTORS.rst``.
+
+.. _contributing_issues:
+
+Bug reports and feature requests
+================================
+
+Bug reports and feature requests can be submitted in the form of a `GitHub issue <https://github.com/JakobGM/astrality/issues/new>`_.
+
+
+.. _contributing_documentation:
+
+Improving the documentation
+===========================
+
+If find something you would like to improve in `the documentation <https://astrality.readthedocs.io/en/latest/index.html>`_, follow these steps:
+
+* Navigate to the page that you would like to edit on https://astrality.readthedocs.io.
+* Press the "Edit on GitHub" link in the upper right corner.
+* Press the "pencil" edit icon to the right of the "History" button.
+* Make the changes you intended.
+* Write a title and description for your change on the bottom of the page.
+* Select the radio button marked as: "Create a new branch for this commit and start a pull request".
+* Press "Propose file change".
+
+The documentation is written in the "RestructuredText" markup language. If this is unfamiliar to you, take a look at `this cheatsheet <https://github.com/ralsina/rst-cheatsheet/blob/master/rst-cheatsheet.rst>`_ for more information.
+
+
+.. _contributing_code:
+
+Contributing code
+=================
+
+Getting up and running
+----------------------
+
+Cloning the repository
+~~~~~~~~~~~~~~~~~~~~~~
+
+First we need to clone the repository. Open your terminal and navigate to the directory you wish to place project directory and run:
+
+.. code-block:: console
+
+    git clone https://github.com/jakobgm/astrality
+    cd astrality
+
+
+Installing python3.6
+~~~~~~~~~~~~~~~~~~~~
+
+Astrality runs on ``python3.6``, so you need to ensure that you have it installed. If you have no specific preferred way of installing software on your computer, you can download and install it from `here <https://www.python.org/downloads/>`_. Alternatively, if you use `brew <https://brew.sh/>`_ on MacOS, you can install it by running:
+
+.. code-block:: console
+
+    brew install python3
+
+Or on ArchLinux:
+
+.. code-block:: console
+
+    sudo pacman -S python
+
+
+Installing dependencies into a virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You should create a separate python3.6 "virtual environment" exclusively for Astrality.
+If this is new to you, take a look at the `official tutorial <https://docs.python.org/3/tutorial/venv.html>`_ for ``venv``.
+A quick summation:
+
+.. code-block:: console
+
+    python3.6 -m venv astrality-env
+    source astrality-env/bin/activate
```

Comment:
Maybe mention that this is for switching into the virtual environment. General flow is to switch into the virtual environment using the source activate command, make/test changes, deactivate to leave the virtual environment. Most shells will have an updated prompt showing the name of the current virtual environment you are working in. You can also run `echo $VIRTUAL_ENV` to see the name (if any) of the virtual environment you are currently in

Reply:
I don't know to what degree we should document an external tool like ``venv``, as there are hundreds of ways to interact with it, ``pyenv``, ``venv``, ``virtualenvwrapper``, and so on...

Well, I added a short description of the workflow anyway, but if we ever get anyone else to contribute, I think its a relatively safe assumption to say that they already know how to use python virtual environments.